artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonAbstractReceiver.java

```diff
@@ -285,13 +282,13 @@ protected void initializeCurrentLargeMessage(Delivery delivery, Receiver receive
    @Override
    public void close(boolean remoteLinkClose) throws ActiveMQAMQPException {
       protonSession.removeReceiver(receiver);
+      clearLargeMessage();
```

Comment:
If remoteLinkClose is false, it possibly shouldnt clean up the message file just yet - more of, or possibly the entire remainder, of the delivery might still be in flight and actually arrive before the peers 'response' detach closure frame arrives (assuming any does). In that case the end of the incomplete message might get processed incorrectly as a corrupted message, potentially leaving another 'large file' unclosed, or worse being considered 'complete' and processed even though the start of it has been thrown away.

Reply:
@gtully can you change the whole method to removeReceiver and clear message upon remoteLinkClose=true?

false here means it was a detach and not a remote close.
I don't remember what a local detach would mean here.. but I guess it's better to be safe on this case.

Reply:
Based on the usage I believe false means it isnt in response to a 'client' asking for it via a detach frame, but is instead a 'local' close initiated on the server either directly (such as when administratively killing a consumer) or indirectly (e.g during connection drop, etc).

In the former case, the server could prvoke sending of a detach(closed=true) to the 'client', but there may still be message data in flight to the server before the client responds to the detach (again, assuming it does reply). If so, it would need to ensure it didnt handle any more payload if it is going to clear the file at this point. Or that it cleared it up later if it doesnt.